### PR TITLE
itest: refactor, add more assertions

### PR DESCRIPTION
**Pull request by guggero**
This PR adds more assertions after refactoring out some common code.

The new assertions discovered the following bugs:
 - https://github.com/lightninglabs/taproot-assets/pull/984
 - https://github.com/lightninglabs/taproot-assets/issues/985

### CODE REVIEW

File: itest/litd_custom_channels_test.go [hunk 1]
```diff
@@ -205,114 +182,17 @@ func testCustomChannels(_ context.Context, net *NetworkHarness,
 	syncUniverses(t.t, charlieTap, dave, erin, fabia, yara)
 	t.Logf("Universes synced between all nodes, distributing assets...")
 
-	// We need to send some assets to Dave, so he can fund an asset channel
-	// with Yara.
 	const (
-		fundingAmount = 50_000
```
Comment (Roasbeef) [inline itest/litd_custom_channels_test.go:211]: Epic diff reduction 💪

File: itest/assets_test.go [hunk 2]
```diff
@@ -913,6 +939,14 @@ func closeAssetChannelAndAssert(t *harnessTest, net *NetworkHarness,
 	t.Logf("Channel closed with txid: %v", closeTxid)
 	t.Logf("Close transaction: %v", spew.Sdump(closeTx))
 
+	// TODO(guggero): Remove this if once the receiver of a channel imports
+	// the proofs correctly as well.
```
Comment (Roasbeef) [inline itest/assets_test.go:943]: For co-op close I presume?

File: itest/litd_custom_channels_test.go [hunk 3]
```diff
@@ -456,14 +457,44 @@ func testCustomChannels(_ context.Context, net *NetworkHarness,
 		false,
 	)
 
-	// For some reason, the channel funding output of the immediately closed
-	// channel is still present in the asset DB, even after we import the
```
Comment (Roasbeef) [inline itest/litd_custom_channels_test.go:640]: Is this fixed by the fix to `SetAssetSpent` in tapd? Or I guess it's the responder co-op close issue? They can't insert the spend as they don't know of the funding output/input.

**Comment by Roasbeef** [comment]
Made this PR to fix the import issue: https://github.com/lightninglabs/taproot-assets/pull/987

Will update this with the latest commit to rerun the itests.

File: itest/litd_custom_channels_test.go (hunk 3 above)
Comment (guggero) [reply itest/litd_custom_channels_test.go:640]: It's fixed by the SetSpend fix.

**Pull request merged**